### PR TITLE
feat: Add activationKeyCodes to configure the keyboard shortcuts

### LINKED ISSUE #584: Make key to start dragging configurable
Right now, space starts dragging & escape stops.

That's ok, but in much of the rest of my application I use 'enter' to activate most controls.

It'd improve my app UX to be able to use that here, or to potentially allow either. Assuming that that's not popular as a default for everybody, I'd like to be able to configure a list of keycodes that should be used to activate dragging.

### PR DESCRIPTION
**Pull request by pimterry**
Fixes #584: you can now set `activationKeyCodes` on the sortable container as an array of the keycodes which should be used to activate/deactivate sorting.

By default it's `[32]` (just space, as before) and I've tested it working in my app with `[32, 13]` (space and enter).

### CODE REVIEW

**Comment by clauderic** [comment]
What do you think of taking a more holistic approach and making all of the keycodes configurable?

Something like:

```jsx
<SortableContainer 
  keyCodes={{
    lift: [32],
    drop: [32],
    cancel: [27],
    up: [38],
    down: [40],
  }}
/>
```

**Comment by pimterry** [comment]
It's interesting, I can definitely see the argument for providing more control. Doing it with one object doesn't play nicely with defaultProps though afaik; you'd have to either use the default set of keys, or redefine all of them, which is a bit awkward. Is there an easy way around that?

**Comment by clauderic** [comment]
Something like this should probably do the trick:

```jsx
const defaultKeyCodes = {
    lift: [32],
    drop: [32],
    cancel: [27],
    up: [38],
    down: [40],
  };

const {keyCodes: customKeyCodes = {}} = this.props;
const keyCodes = {
   ...defaultKeyCodes,
   ...customKeyCodes,
};
```

**Comment by pimterry** [comment]
Sure, that works nicely, done :+1:

File: types/index.d.ts [hunk 1]
```diff
@@ -59,6 +59,13 @@ export interface SortableContainerProps {
   helperClass?: string;
   transitionDuration?: number;
   keyboardSortingTransitionDuration?: number;
+  keyCodes?: {
+    lift?: Array<number>;
```
Comment (clauderic) [inline types/index.d.ts]: nit:
```suggestion
    lift?: number[];
```

File: README.md [hunk 2]
```diff
@@ -102,6 +102,7 @@ There are already a number of great Drag & Drop libraries out there (for instanc
...
+| keyCodes                           | Array<Number>                                             | `{`<br/>`lift: [32],`<br/>`drop: [32],`<br/>`cancel: [27],`<br/>`up: [38, 37],`<br/>`down: [40, 39]`<br/>`}`                                                                                                                        | A object, containing an array of keycodes for each keyboard-accessible action.                                                                                                                                                                                                                                                                   |
```
Comment (clauderic) [inline README.md]:
```suggestion
| keyCodes                           | Array<Number>                                             | `{`<br/>&nbsp;&nbsp;`lift: [32],`<br/>&nbsp;&nbsp;`drop: [32],`<br/>&nbsp;&nbsp;`cancel: [27],`<br/>&nbsp;&nbsp;`up: [38, 37],`<br/>&nbsp;&nbsp;`down: [40, 39]`<br/>`}`                                                                                                                        | An object containing an array of keycodes for each keyboard-accessible action.                                                                                                                                                                                                                                                                   |
```

**Comment by pimterry** [comment]
@clauderic all fixed up :+1:

**Comment by clauderic** [comment]
Released in `1.10.0`

**Pull request merged**